Galaxus.Functional.NUnitExtension.Test/ResultConstraintTest.cs

```diff
@@ -0,0 +1,234 @@
+using NUnit.Framework;
+using NUnit.Framework.Constraints;
+using NUnit.Framework.Internal;
+
+namespace Galaxus.Functional.NUnitExtension.Test;
+
+[TestFixture]
+public class ResultConstraintTest
+{
+    private readonly Result<long, string> _okResult = 42L;
+    private readonly Result<long, string> _errResult = "Failed";
+    private readonly Result<string, string> _okOnSameTypeResult = "ok".ToOk<string, string>();
+
+    private record TestClass(int A, long B);
+
+    private readonly Result<TestClass, string> _complexResult = new TestClass(1, 3).ToOk<TestClass, string>();
+    private const string Description = nameof(ConstraintResult.Description);
+    private const string Status = nameof(ConstraintResult.Status);
+    private const string ActualValue = nameof(ConstraintResult.ActualValue);
+
+    [Test]
+    public void IsResult_Works()
```

Comment:
I'm not a fan of "Check that it works" tests - I prefer specific test cases. Also, the tests in general are quite long and honestly - I don't understand half of them.

This can go hand-in-hand with the other comments I did, but I'd expect something like `Some_PassesWhenObjectIsSome` or `Some_FailsWhenObjectIsSomeWithDifferentValue`. Testing the description (where does it even appear?) should be separate from testing whether the assertion works, too.

I always feel bad doing comments like this one, but in this case I'd rather feel bad for a moment than integrate code I do not understand and could never maintain.

Reply:
Agreed, this Tests name is not the best. 

I had two design goals when writing this extensions. One enabling readable, setence-like tests on Options and Results. The other was to have meaningfull output in case of failure. 

The `Description`-property is the sentence `Expected ...` when a test fails. The `ActualValue`-property contains the `but was ...` phrase. The later can differ between the resolved constraint and the actual string the textwriter outputs. Hence some tests contain a `butWas` condition.

But I agree, one could split the tests further. This one could e.g. split in `IsResult_OnResult_Passes` and `IsResult_HasAMeaningfullDescription`.